### ISSUE TITLE
"TypeError: Cannot read property 'length' of undefined" with no regex matches

### ISSUE THREAD

**Issue by jasonostrander** [Microsoft/BotBuilder#51]
Not sure if this is a bug in the framework or just a lack of proper handling on my part. I'm running the todoBot Node example and getting this error if I just type 'add'. Here's a sample console session:

```
$ node todoBot/textBot.js 
list
You have no tasks.
help
Available commands are:

* *list* - show all tasks (also *show*, *tasks*)
* *new* [your task] - create a new task (also *save*, *create*, *add*)
* *done* [number of task] - finish a task, you can get the number of a task from *list* (also *delete", *finish*, *remove*)
new test
Created a new task no. 1: test
new
Session Error: Cannot read property 'length' of undefined
events.js:154
      throw er; // Unhandled 'error' event
      ^

TypeError: Cannot read property 'length' of undefined
    at /Users/j.ostrander/open-source/BotBuilder/Node/lib/dialogs/CommandDialog.js:46:44
    at Array.forEach (native)
    at CommandDialog.replyReceived (/Users/j.ostrander/open-source/BotBuilder/Node/lib/dialogs/CommandDialog.js:45:33)
    at Session.routeMessage (/Users/j.ostrander/open-source/BotBuilder/Node/lib/Session.js:174:24)
    at next (/Users/j.ostrander/open-source/BotBuilder/Node/lib/Session.js:30:23)
    at Session.dispatch (/Users/j.ostrander/open-source/BotBuilder/Node/lib/Session.js:40:9)
    at /Users/j.ostrander/open-source/BotBuilder/Node/lib/bots/TextBot.js:108:17
    at /Users/j.ostrander/open-source/BotBuilder/Node/lib/bots/TextBot.js:138:21
    at MemoryStorage.get (/Users/j.ostrander/open-source/BotBuilder/Node/lib/storage/Storage.js:9:13)
    at TextBot.getData (/Users/j.ostrander/open-source/BotBuilder/Node/lib/bots/TextBot.js:132:35)```

**Comment by Stevenic** [comment]
thanks... I'll look into the issue.

**Comment by Stevenic** [comment]
We just published v0.7.0 which includes a fix for this.
